extensions/pkg/controller/worker/genericactuator/actuator_reconcile.go

```diff
@@ -121,6 +123,15 @@ func (a *genericActuator) Reconcile(ctx context.Context, log logr.Logger, worker
 		return newError
 	}
 
+	// Update the worker status with the worker pool hash for in-place update worker pools.
+	// If we had reached this point, we can safely say that AutoInPlaceUpdate worker pools have been updated.
+	// Hence we can update their hash in the status.
+	// But for ManualInPlaceUpdate worker pools, we need to check if all the machine deployments for the worker pool are updated.
+	// The worker controller is triggered when a machine deployment is updated, so this can also happen later.
```

Comment:
```suggestion
	// The worker controller is triggered when a machine is updated, so this can also happen later.
```
The controller now acts on Machine updates right?